packages/interledger/src/accounts/service.ts

```diff
@@ -932,17 +932,39 @@ export class AccountsService implements AccountsServiceInterface {
 
     const transfers: TwoPhaseTransfer[] = []
 
-    if (sourceAccount.asset.code === destinationAccount.asset.code) {
-      if (destinationAmount && sourceAmount !== destinationAmount) {
-        return TransferError.InvalidDestinationAmount
-      }
+    if (
+      sourceAccount.asset.code === destinationAccount.asset.code &&
+      sourceAccount.asset.scale === destinationAccount.asset.scale
+    ) {
       transfers.push({
         id: randomId(),
         sourceBalanceId: sourceAccount.balanceId,
         destinationBalanceId: destinationAccount.balanceId,
-        amount: sourceAmount,
+        amount:
+          destinationAmount && destinationAmount < sourceAmount
```

Comment:
maybe `Math.min(sourceAmount, destinationAmount ?? Infinity)` would be clearer?

Reply:
Yeah, I had tried `Math.min` but it sounds like it won't work for `BigInt`s

Reply:
ah, right, nevermind then